app/models/user.rb

```diff
@@ -239,15 +236,12 @@ def one_signal_player_ids
   end
 
   def add_ip(new_ip)
-    unless ip_addresses.include?(new_ip)
-      ips = [new_ip, *ip_addresses].compact.first(PAST_IPS_LIMIT)
-      update_attribute(:ip_addresses, ips)
-    end
-    ip_addresses
+    UserIpaddress.where(user: self, ip_address: new_ip).first_or_create
```

Comment:
This could be `ip_addresses.where(ip_address: new_ip).first_or_create` since the association will provide `user: self` automagically